### PR TITLE
Improve containerd client unit tests

### PR DESCRIPTION
**Pull request by daniel-milchev**
[#51] Improve containerd client unit tests
Signed-off-by: Daniel Milchev fixed-term.daniel.milchev@bosch.io

### CODE REVIEW

File: containerm/ctr/ctrd_container_create_opts_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		})
+	}
+}
+
+func TestWithSpecOpts(t *testing.T) {
+
+	tests := map[string]struct {
+		container *types.Container
+		image     containerd.Image
+		execRoot  string
```
Comment (dimitar-dimitrow) [inline containerm/ctr/ctrd_container_create_opts_test.go:73]: _image_ and _execRoot_ are always the same. Remove them from the test map and create variables and use them when invoking _WithSpecOpts_.

File: containerm/ctr/ctrd_container_create_opts_test.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	}{
+		"test_privileged": {
+			&types.Container{
+				Config: &types.ContainerConfiguration{
+					Cmd: []string{"test"},
+					Env: []string{"test"},
+				},
```
Comment (dimitar-dimitrow) [inline containerm/ctr/ctrd_container_create_opts_test.go:80]: Move in a separate sub-test the case where config is different from nil.

File: containerm/ctr/ctrd_container_create_opts_test.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+			},
+			containerd.NewImage(&containerd.Client{}, images.Image{}),
+			"/tmp/test",
+		},
+		"test_extra_capabilities": {
+			&types.Container{
+				Config: &types.ContainerConfiguration{
+					Cmd: []string{"test"},
+					Env: []string{"test"},
+				},
```
Comment (dimitar-dimitrow) [inline containerm/ctr/ctrd_container_create_opts_test.go:93]:
```suggestion
```

**Review by dimitar-dimitrow — CHANGES_REQUESTED** [review]

File: containerm/ctr/ctrd_container_create_opts_test.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+				HostConfig: &types.HostConfig{
+					ExtraCapabilities: []string{"CAP_NET_ADMIN"},
+				},
+			},
+			containerd.NewImage(&containerd.Client{}, images.Image{}),
+			"/tmp/test",
+		},
+	}
+	for name, tests := range tests {
```
Comment (dimitar-dimitrow) [inline containerm/ctr/ctrd_container_create_opts_test.go:102]:
```suggestion
	for name, test := range tests {
```

File: containerm/ctr/ctrd_container_create_opts_test.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2021 Contributors to the Eclipse Foundation
```
Comment (k-gostev) [inline containerm/ctr/ctrd_container_create_opts_test.go:1]: What year is it?

File: containerm/ctr/ctrd_container_create_opts_test.go [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package ctr
+
+import (
+	"testing"
+
+	"github.com/containerd/containerd"
```
Comment (k-gostev) [inline containerm/ctr/ctrd_container_create_opts_test.go:18]: imports should be formatted in three groups:
```
import (
<system imports>

<current module libraries (*/eclipse-kanto/container-management/*)>

<other>
)

File: containerm/ctr/ctrd_container_create_opts_test.go [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+				HostConfig: &types.HostConfig{
+					ExtraCapabilities: []string{"CAP_NET_ADMIN"},
+				},
+			},
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			got := WithSpecOpts(test.container, containerd.NewImage(&containerd.Client{}, images.Image{}), "/tmp/test")
+			testutil.AssertNotNil(t, got)
```
Comment (k-gostev) [inline containerm/ctr/ctrd_container_create_opts_test.go:95]: `got` variable is not needed

File: containerm/ctr/ctrd_client_test.go [hunk 8]
```diff
@@ -569,13 +576,62 @@ func TestAttachContainer(t *testing.T) {
...
+		},
+	}
 
-	actualError := testClient.AttachContainer(ctx, testCtr, &streams.AttachConfig{})
-	testutil.AssertError(t, expectedError, actualError)
+	for testName, testCase := range tests {
+		t.Run(testName, func(t *testing.T) {
+			expectedError := testCase.mockExec()
+			actualError := testClient.AttachContainer(ctx, testCtr, testCase.attachConfig)
+			testutil.AssertError(t, expectedError, actualError)
```
Comment (k-gostev) [inline containerm/ctr/ctrd_client_test.go:632]: this can be one line

File: containerm/ctr/ctrd_client_test.go [hunk 9]
```diff
@@ -569,13 +576,62 @@ func TestAttachContainer(t *testing.T) {
...
+	tests := map[string]struct {
+		attachConfig *streams.AttachConfig
+		mockExec     func() error
+	}{
+		"test_containerIO_nil": {
+			attachConfig: attachConfig,
+			mockExec: func() error {
+				mockIoMgr.EXPECT().GetIO(testCtr.ID).Return(nil)
+				mockIoMgr.EXPECT().InitIO(testCtr.ID, testCtr.IOConfig.OpenStdin).Return(nil, log.NewErrorf("failed to initialise IO for container ID = test-id"))
+				return log.NewErrorf("failed to initialise IO for container ID = test-id")
```
Comment (k-gostev) [inline containerm/ctr/ctrd_client_test.go:588]:
```suggestion
				return log.NewError("failed to initialise IO for container ID = test-id")
```

**Review by k-gostev — CHANGES_REQUESTED** [review]
I have not into much details, but this is what catch my eye at a glance

**Review by k-gostev — APPROVED** [review]

**Review by dimitar-dimitrow — APPROVED** [review]

**Pull request merged**